allwinner-hal/src/dma/register.rs

```diff
@@ -82,4 +441,183 @@ mod tests {
+        assert_eq!(reg.raw(), 0x2);
+        
+        // Test DMA MCLK circuit gating
+        let reg = reg.disable_dma_mclk_circuit_gating();
+        assert!(!reg.is_dma_mclk_circuit_gating_enabled());
+        assert_eq!(reg.raw(), 0x6);
+        
+        // Test init recommended
+        let reg = AutoGating::init_recommended();
+        assert_eq!(reg.raw(), 0x4);
+        assert!(!reg.is_dma_mclk_circuit_gating_enabled());
+        assert!(reg.is_dma_common_circuit_gating_enabled());
+        assert!(reg.is_dma_chan_circuit_gating_enabled());
+        
+        // Test bit mask
+        let reg = AutoGating::from_raw(0xFFFFFFFF);
+        assert_eq!(reg.raw(), 0x7);
+    }
+
+    #[test]
+    fn test_status() {
+        let reg = Status::from_raw(0x8000FFFF);
+        assert_eq!(reg.raw(), 0x8000FFFF);
+        assert!(!reg.is_mbus_fifo_empty());
+        
+        // Test DMA channels status
+        assert_eq!(reg.dma_channels_status(), 0xFFFF);
+        
+        // Test individual channel status
+        for ch in 0..16 {
+            assert!(reg.is_dma_channel_busy(ch));
+            assert!(!reg.is_dma_channel_idle(ch));
+        }
+        
+        // Test specific channels
+        let reg = Status::from_raw(0x0001);
+        assert!(reg.is_dma_channel_busy(0));
+        assert!(reg.is_dma_channel_idle(1));
+        
+        let reg = Status::from_raw(0x8000);
+        assert!(reg.is_dma_channel_busy(15));
+        assert!(reg.is_dma_channel_idle(0));
+        
+        // Test empty status
+        let reg = Status::from_raw(0x0);
+        assert_eq!(reg.dma_channels_status(), 0x0);
+        for ch in 0..16 {
+            assert!(!reg.is_dma_channel_busy(ch));
+            assert!(reg.is_dma_channel_idle(ch));
+        }
+    }
+
+    #[test]
+    #[should_panic(expected = "Channel must be 0-7 for IrqEnable0")]
+    fn test_irq_enable0_invalid_channel() {
+        let reg = IrqEnable0::from_raw(0x0);
+        reg.enable_interrupt(8, InterruptType::HalfPackage);
+    }
+
+    #[test]
+    #[should_panic(expected = "Channel must be 8-15 for IrqEnable1")]
+    fn test_irq_enable1_invalid_channel() {
+        let reg = IrqEnable1::from_raw(0x0);
+        reg.enable_interrupt(7, InterruptType::HalfPackage);
+    }
+
+    #[test]
+    #[should_panic(expected = "Channel must be 0-7")]
+    fn test_irq_pending0_invalid_channel() {
+        let reg = IrqPending0::from_raw(0x0);
+        reg.if_irq_pending(8, InterruptType::HalfPackage);
+    }
+
+    #[test]
+    #[should_panic(expected = "Channel must be 8-15")]
+    fn test_irq_pending1_invalid_channel() {
+        let reg = IrqPending1::from_raw(0x0);
+        reg.if_irq_pending(75, InterruptType::HalfPackage);
+    }
+
+    #[test]
+    #[should_panic(expected = "Channel number must be 0-15")]
+    fn test_status_invalid_channel() {
+        let reg = Status::from_raw(0x0);
+        reg.is_dma_channel_busy(16);
+    }
+
+}
```

Comment:
```suggestion
    }
}

```
Add a new line in the end of each file.